### PR TITLE
Add option T1 for N4 module

### PR DESCRIPTION
**Pull request by ThoumyreStanislas**
modification of the N4 module to process the T1 image and renaming of inputs and outputs to make them more generic.

### CODE REVIEW

**Review by arnaudbore — APPROVED** [review]
LGTM

**Review by gagnonanthony — APPROVED** [review]
Looks good!

File: modules/nf-scil/preproc/n4/meta.yml [hunk 1]
```diff
@@ -18,19 +18,19 @@ input:
...
       type: file
-      description: Nifti image file - b0
+      description: Nifti image file to use as reference
       pattern: "*.{nii,nii.gz}"
 
-  - b0_mask:
+  - ref_mask:
       type: file
-      description: Nifti image file mask
+      description: Nifti image file mask to use as reference
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/n4/meta.yml:33]: Maybe `for the reference` is a better wording here ?

File: modules/nf-scil/preproc/n4/meta.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@ output:
         Groovy Map containing sample information
         e.g. `[ id:'test', single_end:false ]`
 
-  - dwi:
+  - image:
       type: file
       description: DWI image N4 corrected
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/n4/meta.yml:45]:
```suggestion
       description: N4 corrected image
```

File: modules/nf-scil/preproc/n4/main.nf [hunk 3]
```diff
@@ -22,22 +22,30 @@ process PREPROC_N4 {
...
+        spacing=\$(mrinfo -spacing $ref | tr " " "\\n" | sort -n | tail -1)
+        knot_spacing=\$(echo "\$spacing/$bspline_knot_per_voxel" | bc -l)
 
-    N4BiasFieldCorrection -i $b0\
-        -o [${prefix}__b0_n4.nii.gz, bias_field_b0.nii.gz]\
-        -c [300x150x75x50, 1e-6] -v 1\
-        -b [\${knot_spacing}, 3] \
-        -s $shrink_factor
+        N4BiasFieldCorrection -i $ref\
+            -o [${prefix}__ref_n4.nii.gz, bias_field_ref.nii.gz]\
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/n4/main.nf:36]: This gets included in the `image` output of the module. Either change the name here or specify the glob for the output further.

I am not against outputting this, but it should be done in another optional output channel.

File: tests/modules/nf-scil/preproc/n4/test.yml [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+- name: preproc n4 test_preproc_n4_dwi
+  command: nextflow run ./tests/modules/nf-scil/preproc/n4 -entry test_preproc_n4_dwi -c ./tests/config/nextflow.config
   tags:
     - preproc
     - preproc/n4
   files:
     - path: output/preproc/test__dwi_n4.nii.gz
-      md5sum: 1068c745054b73e4cd836100d34c54a7
+      md5sum: 617da9e7a8783bf0180db0cfdf4a6132
+    - path: output/preproc/test__ref_n4.nii.gz
```
Comment (AlexVCaron) [inline tests/modules/nf-scil/preproc/n4/test.yml:9]: This is the extra output that could pose problem

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Very nice improvment ! A few comments, then GTG !

**Review by AlexVCaron — APPROVED** [review]

**Pull request merged**